_integration-test/run.py

```diff
@@ -0,0 +1,197 @@
+        follow_redirects=True,
+        data={
+            "op": "login",
+            "username": TEST_USER,
+            "password": TEST_PASS,
+            "csrfmiddlewaretoken": login_csrf_token,
+        },
+        headers={"Referer": f"{SENTRY_TEST_HOST}/auth/login/sentry/"},
+    )
+    assert login_response.status_code == 200
+    yield (client, login_response)
+
+
+def test_initial_redirect():
+    initial_auth_redirect = httpx.get(SENTRY_TEST_HOST, follow_redirects=True)
+    assert initial_auth_redirect.url == f"{SENTRY_TEST_HOST}/auth/login/sentry/"
+
+
+def test_login(client_login):
+    client, login_response = client_login
+    assert '"isAuthenticated":true' in login_response.text
+    assert '"username":"test@example.com"' in login_response.text
+    assert '"isSuperuser":true' in login_response.text
+    assert login_response.cookies["sc"] is not None
+    # Set up initial/required settings (InstallWizard request)
+    client.headers.update({"X-CSRFToken": login_response.cookies["sc"]})
+    response = client.put(
+        f"{SENTRY_TEST_HOST}/api/0/internal/options/?query=is:required",
+        follow_redirects=True,
+        headers={"Referer": SENTRY_TEST_HOST},
+        data={
+            "mail.use-tls": False,
+            "mail.username": "",
+            "mail.port": 25,
+            "system.admin-email": "test@example.com",
+            "mail.password": "",
+            "system.url-prefix": SENTRY_TEST_HOST,
+            "auth.allow-registration": False,
+            "beacon.anonymous": True,
+        },
+    )
+    assert response.status_code == 200
+
+
+def test_receive_event(client_login):
+    event_id = None
+    client, _ = client_login
+    with sentry_sdk.init(dsn=get_sentry_dsn(client)):
+        event_id = sentry_sdk.capture_exception(Exception("a failure"))
+    assert event_id is not None
+    response = poll_for_response(
+        f"{SENTRY_TEST_HOST}/api/0/projects/sentry/internal/events/{event_id}/", client
+    )
+    response_json = json.loads(response.text)
+    assert response_json["eventID"] == event_id
+    assert response_json["metadata"]["value"] == "a failure"
+
+
+def test_cleanup_crons_running():
+    docker_services = subprocess.check_output(
+        [
+            "docker",
+            "compose",
+            "--ansi",
+            "never",
+            "ps",
+            "-a",
+        ],
+        text=True
+    )
+    pattern = re.compile(r'(\-cleanup\s+running)|(\-cleanup[_-].+\s+Up\s+)', re.MULTILINE)
+    cleanup_crons = pattern.findall(docker_services)
+    assert len(cleanup_crons) > 0
+
+
+def test_custom_cas():
+    subprocess.run(["./_integration-test/custom-ca-roots/setup.sh"], check=True)
+    subprocess.run(
+        ["docker", "compose", "--ansi", "never", "run", "--no-deps", "web", "python3", "/etc/sentry/test-custom-ca-roots.py"], check=True
+    )
+    subprocess.run(["./_integration-test/custom-ca-roots/teardown.sh"], check=True)
```

Comment:
should probably be `try:` / `finally:` so the teardown runs?  or move the setup/teardown to a `yield`ing fixture

Reply:
good point, added